week1/citibike.R

```diff
@@ -23,26 +23,52 @@ trips <- mutate(trips, gender = factor(gender, levels=c(0,1,2), labels = c("Unkn
 ########################################
 
 # count the number of trips (= rows in the data frame)
-
+nrow(trips)
 # find the earliest and latest birth years (see help for max and min to deal with NAs)
-
+#reference https://www.datasciencemadesimple.com/get-maximum-value-of-a-column-in-r-2/
+#https://stackoverflow.com/questions/652136/how-can-i-remove-an-element-from-a-list
+max(trips$birth_year)
+trips$birth_year[trips$birth_year != "\\N"] %>% min()
 # use filter and grepl to find all trips that either start or end on broadway
-
+#source https://stackoverflow.com/questions/34914173/using-grepl-to-extract-a-value-from-text-across-multiple-columns
+filter(trips, grepl('.*Broadway.*', start_station_name) | grepl('.*Broadway.*', end_station_name))
```

Comment:
This regex works but you don't actually need the .*s, grepl matches on substrings and not the full string like some other regex matchers in other languages. This guy runs more like the re.search in python rather than re.match (which iirc must match the start of the string)